src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/VmrForwardFlower.cs

```diff
@@ -558,16 +558,14 @@ private async Task<bool> RecreatePreviousFlowAndApplyBuild(
     {
         _logger.LogInformation("Failed to create PR branch because of a conflict. Re-creating the previous flow..");
 
-        (Codeflow lastLastFlow, _, _) = await GetLastFlowsAsync(mapping, sourceRepo, currentIsBackflow: true);
-
         // Find the BarID of the last flown repo build
         RepositoryRecord previouslyAppliedRepositoryRecord = _sourceManifest.GetRepositoryRecord(mapping.Name);
         Build previouslyAppliedBuild;
         if (previouslyAppliedRepositoryRecord.BarId == null)
         {
             // If we don't find the previously applied build, we'll just use the previously flown sha to recreate the flow
             // We'll apply a new build on top of this one, so the source manifest will get updated anyway
-            previouslyAppliedBuild = new(-1, DateTimeOffset.Now, 0, false, false, lastLastFlow.SourceSha, [], [], [], []);
+            previouslyAppliedBuild = new(-1, DateTimeOffset.Now, 0, false, false, lastFlow.RepoSha, [], [], [], []);
```

Comment:
I think this makes sense because this scenario only happens when we have two forward flows in a row, where the first one had manual changes in the PR.
This shouldn't happen if a backflow happened in the meantime

Reply:
Maybe you can use `SourceSha`?

Reply:
that would mean if the lastFlow was a backflow, we'd pass a VMR sha, and then these two if's wouldn't have equivalent outcomes?

Reply:
You're right. I think we have a bug there. Maybe we need to load the right build based on the last flow. And that can be either from VMR's `source-manifest.json` or from `V.D.xml` in the repo. Because the build started in one side or another